apps/deepl_integration/handlers.py

```diff
@@ -651,18 +652,42 @@ def save_data(
             page_count=page_count,
             text_extraction_id=text_extraction_id,
         )
-        # Save extracted images as LeadPreviewImage instances
+        # Save extracted images as LeadPreviewAttachment instances
         # TODO: The logic is same for unified_connector leads as well. Maybe have a single func?
-        image_base_path = f'{lead.pk}'
+
+        attachement_base_path = f'{lead.pk}'
         for image_uri in images_uri:
-            lead_image = LeadPreviewImage(lead=lead)
-            image_obj = RequestHelper(url=image_uri, ignore_error=True).get_file()
-            if image_obj:
-                lead_image.file.save(
-                    os.path.join(image_base_path, os.path.basename(urlparse(image_uri).path)),
-                    image_obj
+            for image in image_uri['images']:
+                lead_attachement = LeadPreviewAttachment(lead=lead)
+                image_obj = RequestHelper(url=image, ignore_error=True).get_file()
+                if image_obj:
+                    lead_attachement.file.save(
+                        os.path.join(attachement_base_path, os.path.basename(urlparse(image).path)),
+                        image_obj
```

Comment:
```suggestion
```suggestion
                        os.path.join(
                            attachement_base_path,
                            os.path.basename(urlparse(image).path),
                         ),
                         image_obj,
```
```